demo.html

```diff
@@ -41,6 +41,32 @@ <h2>General usage example</h2>
 
     <hr />
 
+    <h2>Media query</h2>
+    <p>
+      Media query matching is reactive
+      using the media query list's change event,
+      so it will react to viewport 
+      and non-viewport media query changes.
+    </p>
+
+    <show-when id="change" has-media="(orientation: portrait)" hidden>
+      The viewport is portrait. Click to change.
+    </show-when>
+    <script>
+      change.addEventListener('click', ()=>{
+        change.setAttribute('has-media', '(orientation: landscape)');
+        change.innerText = 'The media query is now `(orientation: landscape)`';
+      })
+    </script>
```

Comment:
I think this could use some clarification: is it reactive to the media changing, the attribute changing, or both? I think the latter was already true - and the new feature is reacting to changes in media? In which case, I don't think the click event adds much here. We should just show both, and suggest resizing the window to see it change?

Reply:
You're correct that it was already reactive to the attribute changing, and now it is reactive to both. Because we are attaching an event listener to a MediaQueryList for the has-media attribute's value, I wanted a way to test that it was listening to the updated value. Although, in this example the values are inverses (and so would be triggered at the same time), so it doesn't explain that well.

The other listeners don't need to react to the attribute's value changing (we listen for all hash changes, for instance).

I think you're right that this isn't a particularly helpful example, and was more of just a test case, so I'll switch back to landscape/portrait.